jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -266,27 +286,28 @@ private static void addIncludePaths(@NotNull GeneralCommandLine commandLine, @Nu
   @NotNull
   private static List<String> getErlangModulePaths(@NotNull ErlangTarget target,
                                                    @NotNull CompileContext context,
-                                                   @NotNull List<String> dirtyFilePaths,
                                                    boolean isTest) {
-    List<ErlangFileDescriptor> fileDescriptors = getErlangFileDescriptorFromConfig(target, context, isTest);
-    return fileDescriptors != null ?
-           getSortedErlangModulePathsToCompile(fileDescriptors, dirtyFilePaths) :
-           getErlangModulePathsDefault(target, isTest);
+    List<String> erlangBuilderDirtyFilesInfo = getErlangModulePathsFromTarget(target, context, isTest);
+    return erlangBuilderDirtyFilesInfo != null ? erlangBuilderDirtyFilesInfo : getErlangModulePathsDefault(target, isTest);
   }
 
-  @NotNull
-  private static List<String> getSortedErlangModulePathsToCompile(@NotNull List<ErlangFileDescriptor> sortedModuleDescriptors,
-                                                                  @NotNull List<String> dirtyModules) {
-    SortedModuleDependencyGraph semiGraph = new SortedModuleDependencyGraph(sortedModuleDescriptors);
-    GraphGenerator<Node> graph = GraphGenerator.create(semiGraph);
-    markDirtyNodes(semiGraph.getNodesByName(dirtyModules), graph);
-    return ContainerUtil.mapNotNull(semiGraph.getNodes(), new Function<Node, String>() {
-      @Nullable
-      @Override
-      public String fun(Node node) {
-        return node.myDirty && FileUtilRt.extensionEquals(node.myErlangModulePath, "erl") ? node.myErlangModulePath : null;
-      }
-    });
+  @Nullable
+  private static List<String> getErlangModulePathsFromTarget(@NotNull ErlangTarget target,
+                                                             @NotNull CompileContext context,
+                                                             boolean isTests) {
+    ErlangModuleBuildOrder buildOrder = target.getBuildOrder();
+    if (buildOrder == null) {
+      String message = "Erlang module " + target.getModule().getName() + " will be fully rebuilt.";
+      LOG.debug(message);
+      context.processMessage(new CompilerMessage(NAME, BuildMessage.Kind.WARNING, message));
```

Comment:
This is not the place where decision to fully rebuild this module is made - move this logging and reporting to where the responsible code is.